pyfr/integrators/std/controllers.py

```diff
@@ -77,7 +77,12 @@ def advance_to(self, t):
 
         while self.tcurr < t:
             # Decide on the time step
-            dt = max(min(t - self.tcurr, self._dt), self.dtmin)
+            if self.tcurr + 2*self._dt < t < self.tcurr + 3*self._dt:
```

Comment:
I don't think we ever had any problems with std controller and the existing timestep strategy. std integrators can take a very small step and it doesn't cause any issues as far as I know

Reply:
Unless you had any issues I think it is better to stick with the current approach

Reply:
I forgot to remove this, I was testing the code for `std` before moving to `dual`.

Reply:
Apart from a small bug here, I guess this stays as it is? 

If we do want to gradually change dt here, since `self._dt` is changes with local controller here, we cannot copy-paste the same idea for implicit here. One option would be to assume the self._dt about 10 timesteps prior to desired time to be set as a constant, untill the desired time is reached.